test/e2e/cronjob_test.go

```diff
@@ -57,6 +60,47 @@ func TestSourceCronJob(t *testing.T) {
 	out, err := test.getResourceFieldsWithJSONPath("cronjobsource", "testcronjobsource2", jpSinkRefNameInSpec)
 	assert.NilError(t, err)
 	assert.Equal(t, out, "testsvc1")
+
+	t.Log("verify event messages from cronjob source")
+	mymsg := "This is a message from cronjob."
+	test.serviceCreateEventDisplay(t, r, "displaysvc")
+	test.cronJobSourceCreate(t, r, "testcronjobsource3", "*/1 * * * *", mymsg, "svc:displaysvc")
+	err = test.verifyDeventDisplayLogs(t, "displaysvc", mymsg)
+	assert.NilError(t, err)
+}
+
+func (test *e2eTest) verifyDeventDisplayLogs(t *testing.T, svcname string, message string) error {
+	var (
+		retries int
+		err     error
+		out     string
+	)
+
+	selectorstr := fmt.Sprintf("serving.knative.dev/service=%s", svcname)
+	for retries < 5 {
+		out, err = kubectl{}.Run("logs", "-l", selectorstr, "-c", "user-container")
+		if err != nil {
+			t.Logf("error happens at kubectl logs -l %s : %v", selectorstr, err)
+		} else if err == nil && strings.Contains(out, message) {
+			break
+		} else {
+			t.Logf("return from kubectl logs -l %s : %s", selectorstr, out)
+		}
+		retries++
+		time.Sleep(2 * time.Minute)
+	}
+
+	if retries == 5 {
+		return fmt.Errorf("Expected log incorrect after retry 5 times. Expecting to include:\n%s\n Instead found:\n%s\n", message, out)
+	} else {
+		return nil
+	}
+}
+
+func (test *e2eTest) serviceCreateEventDisplay(t *testing.T, r *KnRunResultCollector, serviceName string) {
```

Comment:
Why is this not a public `Test...` function?

Reply:
It's same as `serviceCreate` function defined in `service_test.go`
```
func (test *e2eTest) serviceCreateDuplicate(t *testing.T, r *KnRunResultCollector, serviceName string)
```
Although it's not public, but you can use this function under any e2e tests.